app/controllers/application_controller.rb

```diff
@@ -1852,10 +1857,31 @@ def identify_tl_or_perf_record
     identify_record(params[:id], controller_to_model)
   end
 
-  def assert_privileges(*features)
-    if features.none? { |feature| role_allows?(:feature => feature) }
-      raise MiqException::RbacPrivilegeException, _('The user is not authorized for this task or item.')
-    end
+  def assert_privileges(*features, any: false)
+    msg = "Features checked: #{features.join(', ')}"
+
+    pass =
+      if any
+        role_allows?(:feature => features.first, :any => true)
+      else
+        features.any? { |feature| role_allows?(:feature => feature) }
+      end
+
+    log_privileges(pass, msg)
+    raise MiqException::RbacPrivilegeException, _('The user is not authorized for this task or item.') unless pass
+  end
+
+  def log_privileges(pass, details = nil)
+    # This is called with or without a current user and possibly a fake request such as in test.
+    username    = current_userid rescue nil
+    role_name   = current_user.miq_user_role.name rescue nil
+    http_method = request.respond_to?(:request_method) ? request.request_method : nil
+    path        = request.respond_to?(:filtered_path)  ? request.filtered_path  : nil
+    request_id  = request.respond_to?(:request_id)     ? request.request_id     : nil
```

Comment:
Look at the example output in the description... You'll see request like this where a single request cuts more than one audit log entry as some could check multiple product features:

```
[----] I, [2023-01-27T10:07:33.368293 #17089:5a398]  INFO -- audit: <AuditSuccess> Username [joe], Role [EvmRole-user], Request [2009ddce-eaa3-46fb-bc92-ea5fd67b3160], Method [GET], Path [/dashboard/widget_chart_data/6] Action: widget_chart_data
[----] I, [2023-01-27T10:07:33.368664 #17089:5a398]  INFO -- audit: <AuditSuccess> Username [joe], Role [EvmRole-user], Request [2009ddce-eaa3-46fb-bc92-ea5fd67b3160], Method [GET], Path [/dashboard/widget_chart_data/6] Features checked: dashboard_view
```